src/exports/arcdps.rs

```diff
@@ -0,0 +1,497 @@
+];
+
+static KEYBINDS: RwLock<BTreeMap<Control, KeybindChange>> = RwLock::new(BTreeMap::new());
+
+pub(crate) fn cb_extras_keybind(changed: KeybindChange) {
+    if !INTERESTING_BINDS.contains(&changed.control) {
+        return
+    }
+
+    let mut kb = match KEYBINDS.write() {
+        Ok(kb) => kb,
+        Err(_) => {
+            log::warn!("Keybinds poisoned?");
+            return
+        },
+    };
+    kb.insert(changed.control, changed);
+}
+
+#[cfg(feature = "extension-arcdps-codegen")]
+pub(crate) fn cb_extras_squad_update(members: UserInfoIter) {
+    use crate::receive_squad_update;
+
+    receive_squad_update(members)
+}
+
+pub fn available() -> bool {
+    RUNTIME_AVAILABLE.load(Ordering::Relaxed)
+}
+
+pub fn extras_available() -> bool {
+    EXTRAS_AVAILABLE.load(Ordering::Relaxed)
+}
+
+const NO_EXPORT: &'static str = "arcdps export missing";
+
+pub fn addon_dir() -> RuntimeResult<Option<PathBuf>> {
+    if !available() {
+        return Ok(None)
+    }
+
+    if !arc::has_e0_config_path() {
+        return Err(NO_EXPORT)
+    }
+    let mut path = arcdps::exports::config_path()
+        .ok_or("Unknown arcdps config dir")?;
+    // remove ini leaf from path...
+    if !path.pop() {
+        return Err("Incomplete config path")
+    }
+
+    let in_addons = path.file_name() == Some(OsStr::new("arcdps"))
+        || path.parent().and_then(|p| p.file_name()) == Some(OsStr::new("addons"));
+    if in_addons {
+        path.pop();
+    }
+
+    path.push(exports::ADDON_DIR_NAME);
+    Ok(Some(path))
+}
+
+pub fn detect_language() -> RuntimeResult<Option<String>> {
+    if !available() {
+        return Ok(None)
+    }
+
+    let language = game_language().map(game_language_id);
+    Ok(language.map(Into::into))
+}
+
+pub fn mumble_link_ptr() -> RuntimeResult<Option<MumblePtr>> {
+    if !available() {
+        return Ok(None)
+    }
+
+    match mumble_ptr() {
+        Some(ml) => Ok(Some(ml)),
+        None => Err("MumbleLink unavailable"),
+    }
+}
+
+pub fn nexus_link_ptr() -> RuntimeResult<Option<NonNull<NexusLink>>> {
+    if !available() {
+        return Ok(None)
+    }
+
+    Err("NexusLink unavailable")
+}
+
+pub fn rtapi() -> RuntimeResult<Option<RealTimeApi>> {
+    if !available() {
+        return Ok(None)
+    }
+
+    Err("RTAPI unsupported")
+}
+
+pub fn invoke_marker_bind(marker: MarkerType, target: bool, duration_ms: i32) -> RuntimeResult<Option<()>> {
+    if !available() {
+        return Ok(None)
+    }
+
+    let control = match target {
+        true => marker.control_object(),
+        false => marker.control_location(),
+    };
+
+    let binding = {
+        let kb = KEYBINDS.read()
+            .map_err(|_| "keybinds poisoned")?;
+        kb.get(&control).cloned()
+    }.ok_or("unknown keybind")?;
+
+    let KeybindChange { key, mod_shift, mod_alt, mod_ctrl, .. } = binding;
+
+    Err("TODO: invoke marker bind")
```

Comment:
this is where you simulate the keypress in whichever way you like...
- [ ] probably just `PostMessage` the modifiers and key events
- [ ] since this is replacing nexus "invoke async", it's meant to be run on a background thread and sleep between the down+up events
- [ ] please reconsider using `invoke_async` and just switching to manual (non-async) up/down requests and then  `sleep(duration).await` in-between since this is only going to be called from the controller anyway, it doesn't need to spawn a damn thread or worker just to press a button!